### DISCUSSION TITLE
Preloading font referenced in .css files

### DISCUSSION THREAD

**Discussion by phuoccss** [remix-run/remix#4648] [Q&A]
The css plugin here https://github.com/remix-run/remix/pull/4130 ensure all assets, including fonts, referenced in css files are bundled and added to the `_assets` folder.

Does anyone have any tip on getting the url for the bundled hashed font file? I'm trying to preload the font via route module's `links` function, but esbuild throws an error when importing the font file from node_modules.

![image](https://user-images.githubusercontent.com/15166715/202852905-900c4156-e780-4da3-ae6c-c482fb5f8ac5.png)

Perhaps this can even be automated, similar to <Script/> component's modulepreload links.

**Comment by sergiodxa** [comment]
I haven’t tried myself but I would expect the hash in the url to be the same that if you imported the font on your JS code

```ts
import font from "~/fonts/name.woff2"

export let links: LinksFunction = () => {
  return [{ rel: "preload", as: "font", href: font }]
}
```

**Reply by phuoccss** [reply]
You're right, but somehow this doesn't work for font files imported from `node_modules`. esbuild throws an error, as shown in the OP.

Maybe it's a bug with Remix? It should work since `woff2` is supported, as seen [here](https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/loaders.ts). There's no problem importing css files from `node_modules` for eg.

**Reply by sergiodxa** [reply]
That could be a bug, yes, I recommend you to try to create a minimal reproduction example (the basic default app + importing a css from node_modules loading a font) and open an issue with the bug report
